File: Assets/MakeHuman/Icons/KsSoft/Editor/Multilingual/tools/protocol/GmsvMessage.py
from CVariable import CReadVariable,CWriteVariable
from Vector import *
from t_Message import	t_Message
#/==========================================================================
#*!
#	@brief	GmsvMessage
#/
class GmsvMessage:
	#/==========================================================================
	#*!
	#	@brief	Member
	#/
	m_eChatMode = 0	#S8
	m_tMessage = t_Message()	#t_Message
	#/==========================================================================
	#*!
	#	@brief	Constructor
	#/
	def __init__(self):
		self.clear()

	#/==========================================================================
	#*!
	#	@brief	Accessor
	#/
	def clear(self):
		self.m_eChatMode = 0
		self.m_tMessage = t_Message()

	def read(self,cVariable):
		self.m_eChatMode = cVariable.getS8()
		if (not self.m_tMessage.read(cVariable)):
			return False
		return True

	def write(self,cVariable):
		cVariable.putS8(self.m_eChatMode)
		if (not self.m_tMessage.write(cVariable)):
			return False
		return True

File: Assets/MakeHuman/Icons/KsSoft/Plugins/Protocol/GmsvMessage.cs
//==============================================================================================
/*!
	@file  KsSoft Protocol
	
	(counter SJIS string 京)
*/
//==============================================================================================
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

using KS;
//==========================================================================
/*!
	@brief	GmsvMessage
*/
public class GmsvMessage : ISerializable {
	//==========================================================================
	/*!
		@brief	Member
	*/
	public sbyte	m_eChatMode;
	public t_Message	m_tMessage;
	//==========================================================================
	/*!
		@brief	Constructor
	*/
	public GmsvMessage() {
		m_tMessage = new t_Message();
		clear();
	}
	//==========================================================================
	/*!
		@brief	Accessor
	*/
	public void clear() {
		m_eChatMode = 0;
		m_tMessage.clear();
	}
	public bool read(CReadVariable cVariable) {
		try {
			m_eChatMode = cVariable.getS8();
			if (!m_tMessage.read(cVariable)) return false;
		} catch (System.Exception e) {
			Debug.LogError(e);return false;
		}
		return true;
	}
	public bool write(CWriteVariable cVariable) {
		cVariable.put(m_eChatMode);
		if (!m_tMessage.write(cVariable)) return false;
		return true;
	}
};

